File: Scala/NewFizzbuzz.scala
// FizzBuzz in scala
// Author: emilkovacev

object FizzBuzz {
  def main(args: Array[String]): Unit = {
    val A = 3  // multiple: Fizz
    val B = 5  // multiple: Buzz

    def fizzBuzzCalc(value: Int): String = {
      var retVal = ""

      if (value % A == 0) retVal += "Fizz"
      if (value % B == 0) retVal += "Buzz"
      retVal
    }

    for (i <- 1 to 100) println(if (fizzBuzzCalc(i).isEmpty) i else fizzBuzzCalc(i))
  }
}

File: Rust/new_fizzbuzz.rs
--author: @nchiarappa

fn main() {
    for x in 1..=100 {
        match (x % 3, x % 5) {
            (0, 0) => println!("FizzBuzz"),
            (0, _) => println!("Fizz"),
            (_, 0) => println!("Buzz"),
            _ => println!("{}", x),
        }
    }
}